encoding/codecv4/codecv4_test.go

```diff
@@ -0,0 +1,837 @@
+
+	trace := readBlockFromJSON(t, "../testdata/blockTrace_02.json")
+	originalChunk := &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0x820f25d806ddea0ccdbfa463ee480da5b6ea3906e8a658417fb5417d0f837f5c", hash.Hex())
+
+	trace = readBlockFromJSON(t, "../testdata/blockTrace_03.json")
+	originalChunk = &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0x4620b3900e8454133448b677cbb2054c5dd61d467d7ebf752bfb12cffff90f40", hash.Hex())
+
+	trace = readBlockFromJSON(t, "../testdata/blockTrace_04.json")
+	originalChunk = &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0x059c6451e83012b405c7e1a38818369012a4a1c87d7d699366eac946d0410d73", hash.Hex())
+
+	trace = readBlockFromJSON(t, "../testdata/blockTrace_05.json")
+	originalChunk = &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0x854fc3136f47ce482ec85ee3325adfa16a1a1d60126e1c119eaaf0c3a9e90f8e", hash.Hex())
+
+	trace = readBlockFromJSON(t, "../testdata/blockTrace_06.json")
+	originalChunk = &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0x2aa220ca7bd1368e59e8053eb3831e30854aa2ec8bd3af65cee350c1c0718ba6", hash.Hex())
+
+	trace = readBlockFromJSON(t, "../testdata/blockTrace_07.json")
+	originalChunk = &encoding.Chunk{Blocks: []*encoding.Block{trace}}
+	chunk, err = NewDAChunk(originalChunk, 0)
+	assert.NoError(t, err)
+	hash, err = chunk.Hash()
+	assert.NoError(t, err)
+	assert.Equal(t, "0xb65521bea7daff75838de07951c3c055966750fb5a270fead5e0e727c32455c3", hash.Hex())
+}
+
+func TestCodecV4BatchEncode(t *testing.T) {
+	// empty batch
+	batch := &DABatch{Version: uint8(encoding.CodecV4)}
+	encoded := hex.EncodeToString(batch.Encode())
+	assert.Equal(t, "04000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", encoded)
+
+	trace2 := readBlockFromJSON(t, "../testdata/blockTrace_02.json")
+	chunk2 := &encoding.Chunk{Blocks: []*encoding.Block{trace2}}
+	originalBatch := &encoding.Batch{Chunks: []*encoding.Chunk{chunk2}}
+	batch, err := NewDABatch(originalBatch, true /* enble encode */)
```

Comment:
```suggestion
	batch, err := NewDABatch(originalBatch, true /* enable encode */)
```

Reply:
Please update all occurrences

Reply:
fixed in [7fd9672](https://github.com/scroll-tech/da-codec/pull/23/commits/7fd9672293445b11d96e7f5528cfae7b52135e70).